contracts/StablecoinConverter.sol

```diff
@@ -209,20 +215,46 @@ contract StablecoinConverter is EpochTokenLocker {
                 executedSellAmount
             );
         }
-        checkAndOverrideObjectiveValue(uint(tokenConservation[0]));
+        int objectiveValue = utility - disregardedUtility;
+        checkAndOverrideObjectiveValue(objectiveValue);
         grantRewardToSolutionSubmitter(uint(tokenConservation[0]) / 2);
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
-    function getCurrentObjectiveValue() public view returns(uint) {
+    function getCurrentObjectiveValue() public view returns(int) {
         if (previousSolution.batchId == getCurrentBatchId() - 1) {
             return previousSolution.objectiveValue;
         } else {
-            return 0;
+            return MIN_INT;
         }
     }
 
+    function evaluateUtility(uint128 execBuy, uint128 execSell, Order memory order) internal view returns(int) {
+        // Utility = ((execBuyAmt * order.sellAmt - execSellAmt * order.buyAmt) * price.buyToken) / order.sellAmt
+        (uint128 buyAmount, uint128 sellAmount) = getBuySellAmounts(order);
+        return ((execBuy * sellAmount - execSell * buyAmount) * currentPrices[order.buyToken]) / sellAmount;
+    }
+
+    function evaluateDisregardedUtility(uint128 execSell, Order memory order) internal view returns(int) {
+        // |disregardedUtility| = maxUtility - actualUtility
+        // where maxUtility is the utility achieved when execSellAmount == order.sellAmount. That is,
+        // maxUtility = ((execBuyAmt * order.sellAmt - order.sellAmt * order.buyAmt) * price.buyToken) / order.sellAmt
+        //            = (execBuyAmt - order.buyAmount) * price.buyToken <---- This reduction not used in final formula.
+        // So, after simplification
+        (uint128 buyAmount, uint128 sellAmount) = getBuySellAmounts(order);
+        return (((sellAmount - execSell) * currentPrices[order.buyToken]) * buyAmount) / sellAmount;
```

Comment:
Does this take into account if the account's balance would have been able to afford this trade? I think this is a crucial non-trivial requirement for that metric.

Reply:
I believe it does... since `submitSolution` would fail if it didn't.

Reply:
I believe it does not. All checks are done with regards to the volume, submitted as part of the solution if I recall correctly. But your function "just" looks at the order.remainingAmount

Reply:
and `order.remainingAmount` should be up to date with the balance. Unless it is not updated whenever deposits and withdraws are made/requested.

Reply:
If this is not the case, I would argue that this is an unrelated issue to the content of the PR.